### ISSUE TITLE
Build fails on FreeBSD (2.1.27-rc5)

### ISSUE THREAD

**Issue by UnitedMarsupials** [cyrusimap/cyrus-sasl#490]
Trying to compile the code, I get the following error:

```
depbase=`echo allockey.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\ && ../libtool  --tag=CC   --mode=compile cc -DHAVE_CONFIG_H -I. -I..  -fPIC -I../include -I../include  -DOBSOLETE_CRAM_ATTR=1 -DOBSOLETE_DIGEST_ATTR=1  -Wall -W -O2 -pipe -fstack-protector -march=core2  -fno-strict-aliasing -MT allockey.lo -MD -MP -MF $depbase.Tpo -c -o allockey.lo allockey.c &&\
mv -f $depbase.Tpo $depbase.Plo
 : not found
gmake[4]: *** [Makefile:502: allockey.lo] Error 127
gmake[4]: Leaving directory '/spare/usr/ports/security/cyrus-sasl2/work/cyrus-sasl-2.1.27/sasldb'
gmake[3]: *** [Makefile:685: all-recursive] Error 1
```
I think, the ` :` is relying on some kind of `bash`-ism... Using the `--disable-dependency-tracking` I'm able to work-around this problem.

BTW, bash or not, why not just use `${DEPDIR}/$*.Tpo` instead of invoking `sed` for every file?

**Comment by makr** [comment]
No, `:` is not a bashism. It is a standard [POSIX shell builtin](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#colon) for `true`.

**Comment by UnitedMarsupials** [comment]
Yes, `:` is a built-in. But it seem to try to exec ` :` -- with a space at the beginning. Not sure, where _that_ is coming from or how it can work for anyone -- hence the guess, bash works subtly differently in this case.

**Comment by flowerysong** [comment]
This has nothing to do with ```:```.

```depbase=`echo allockey.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\ && ../libtool  --tag=CC   --mode=compile cc -DHAVE_CONFIG_H -I. -I..  -fPIC -I../include -I../include  -DOBSOLETE_CRAM_ATTR=1 -DOBSOLETE_DIGEST_ATTR=1  -Wall -W -O2 -pipe -fstack-protector -march=core2  -fno-strict-aliasing -MT allockey.lo -MD -MP -MF $depbase.Tpo -c -o allockey.lo allockey.c &&\
mv -f $depbase.Tpo $depbase.Plo```

See that ```\ && ../libtool``` there?  That's trying to run ``` ```, which results in ``` : not found```, because ``` ``` is the command that is not found.

**Comment by UnitedMarsupials** [comment]
> See that \ && ../libtool there? That's trying to run , which results in : not found, because is the command that is not found.

Indeed! Still needs fixing then, does not it?

**Comment by dilyanpalauzov** [comment]
I am not sure what file is missing.

If libtool is is missing. it is supposed to be generated by `config.status` called at the end by `./configure`.  Does `./config.status` generate libtool, and the problem is that  ./configure does not call ./config.status?

Does `./configure` report any error (e.g. to STDERR), do you run autoreconf -vf/autogen.sh before calling `./configure`?

If libtool is not missing, use strace to determine which file exactly is missing.

**Comment by ksmurchison** [comment]
Is this still an issue?  Have you tracked down the missing file?

**Comment by UnitedMarsupials** [comment]
I got distracted and decided to wait for the [FreeBSD port](https://www.freshports.org/security/cyrus-sasl2)'s maintainer to update the port...
